zerver/lib/export.py

```diff
@@ -80,29 +82,302 @@ def floatify_datetime_fields(data, table):
+        self.is_seeded = is_seeded
+        self.exclude = exclude
+        self.custom_fetch = custom_fetch
+        self.custom_tables = custom_tables
+        self.concat_and_destroy = concat_and_destroy
+        self.children = [] # type: List[Config]
+
+        if virtual_parent:
+            self.parent = virtual_parent
+        elif normal_parent:
+            self.parent = normal_parent
+        else:
+            self.parent = None
+
+        if self.parent:
+            self.parent.children.append(self)
+
+def export_from_config(response, config, seed_object=None, context=None):
+    table = config.table
+    parent = config.parent
+    model = config.model
+
+    if context is None:
+        context = {}
+
+
+    if table:
+        exported_tables = [table]
+    else:
+        exported_tables = config.custom_tables
+
+    for t in exported_tables:
+        logging.info('Exporting via export_from_config:  %s' % (t,))
+
+    rows = None
+    if config.is_seeded:
+        rows = [seed_object]
+
+    elif config.custom_fetch:
+        config.custom_fetch(
+            response=response,
+            config=config,
+            context=context
+        )
+        if config.custom_tables:
+            for t in config.custom_tables:
+                if t not in response:
+                    raise Exception('Custom fetch failed to populate %s' % (t,))
+
+    elif config.concat_and_destroy:
+        # When we concat_and_destroy, we are working with
+        # temporary "tables" that are lists of records that
+        # should already be ready to export.
+        data = [] # type: List[Record]
+        for t in config.concat_and_destroy:
+            data += response[t]
+            del response[t]
+            logging.info('Deleted temporary %s' % (t,))
+        response[table] = data
+
+    elif config.use_all:
+        query = model.objects.all()
+        rows = list(query)
+
+    elif config.normal_parent:
+        model = config.model
+        parent_ids = [r['id'] for r in response[parent.table]]
+        filter_parms = {config.parent_key: parent_ids}
+        if config.filter_args:
+            filter_parms.update(config.filter_args)
+        query = model.objects.filter(**filter_parms)
+        rows = list(query)
+
+    # Post-process rows (which won't apply to custom fetches/concats)
+    if rows is not None:
+        response[table] = make_raw(rows, exclude=config.exclude)
+        if table in DATE_FIELDS:
+            floatify_datetime_fields(response, table)
+
+    # Now walk our children.  It's extremely important to respect
+    # the order of children here.
+    for child_config in config.children:
+        export_from_config(
+            response=response,
+            config=child_config,
+            context=context,
+        )
+
+def get_realm_config():
+    # type: () -> Config
+    # This is common, public information about the realm that we can share
+    # with all realm users.
+
+    realm = Config(
```

Comment:
would call this local variable `realm_config` throughout the code since we do have a clear meaning for local variables called `realm`